packages/expo-audio/ios/AudioRecordingRequester.swift

```diff
@@ -33,15 +36,22 @@ public class AudioRecordingRequester: NSObject, EXPermissionsRequester {
     ]
   }
 
-  public func requestPermissions(resolver resolve: @escaping EXPromiseResolveBlock, rejecter reject: EXPromiseRejectBlock) {
-    if #available(iOS 17.0, *) {
-      AVAudioApplication.requestRecordPermission { _ in
-        resolve(self.getPermissions())
-      }
-    } else {
-      AVAudioSession.sharedInstance().requestRecordPermission { _ in
-        resolve(self.getPermissions())
+  public func requestPermissions(resolver resolve: @escaping EXPromiseResolveBlock, rejecter reject: @escaping EXPromiseRejectBlock) {
+    typealias PermissionRequestFunction = @convention(c) (AnyObject, Selector, @escaping (Bool) -> Void) -> Void
+
+    let session = AVAudioSession.sharedInstance()
+    guard let method = class_getInstanceMethod(type(of: session), recordPermissionSelector) else {
+      return
```

Comment:
I think we should reject the promise here to avoid leaving the promise in a pending state forever

Reply:
In theory, this should never fail but better to do it just to be safe 👍